src/app/polling-progress/polling-progess.component.ts

```diff
@@ -44,19 +44,18 @@ export class PollingProgessComponent<T> {
       const pollIntervalInSeconds = this.pollIntervalInSeconds();
 
       this.progressBarValue$ = timer(0, updateProgressBarInterval).pipe(
-        tap((value) => {
+        tap(() => {
           if (
-            value * updateProgressBarInterval ===
+            this.milliSecondsSinceLastUpdate() >=
             pollIntervalInSeconds * 1000
           ) {
             this.resource().reload();
           }
         }),
         map(
-          (value) =>
-            ((((value + 1) * 100) / pollIntervalInSeconds) *
-              updateProgressBarInterval) /
-            1000,
+          () =>
+            (100 / (pollIntervalInSeconds * 1000)) *
+            this.milliSecondsSinceLastUpdate(),
```

Comment:
```suggestion
          () =>
            this.milliSecondsSinceLastUpdate() / (pollIntervalInSeconds * 1000) * 100,
```
would look a little more "logical" to my eyes. 

Mixing seconds and milliseconds makes everything a little harder. Maybe one could turn `millisecondsSinceLastUpdate` into `secondsSinceLastUpdate` version then the `* 1000` in two places is not necessary?

Reply:
Done!